app/services/proxy_status/pulp.rb

```diff
@@ -0,0 +1,46 @@
+require 'uri'
+
+module ProxyStatus
+  class Pulp < ::ProxyStatus::Base
+    def initialize(proxy, opts = {})
+      super(proxy, opts, 'Pulp')
+    end
+
+    def storage
+      fetch_proxy_data do
+        api.pulp_storage
+      end
+    end
+
+    def status
+      body = RestClient.get(pulp_url)
+      fail _("Pulp does not appear to be running.") if body.empty?
+      json = JSON.parse(body)
+      json['errors'] = {}
+
+      if json['known_workers'].empty?
+        json['errors']['known_workers'] = _("No pulp workers running.")
+      end
+
+      if json['database_connection'] && json['database_connection']['connected'] != true
+        json['errors']['database_connection'] = _("Pulp database connection issue.")
+      end
+
+      if json['messaging_connection'] && json['messaging_connection']['connected'] != true
+        json['errors']['messaging_connection'] = _("Pulp message bus connection issue.")
+      end
+
+      json
+    end
```

Comment:
I'm good with that approach.

On Tue, Jan 19, 2016 at 7:09 AM, Shlomi Zadok <notifications@github.com>
wrote:

> In app/services/proxy_status/pulp.rb
> <https://github.com/Katello/katello/pull/5716#discussion_r50104629>:
>
> > +      json['errors'] = {}
> > +
> > +      if json['known_workers'].empty?
> > +        json['errors']['known_workers'] = _("No pulp workers running.")
> > +      end
> > +
> > +      if json['database_connection'] && json['database_connection']['connected'] != true
> > +        json['errors']['database_connection'] = _("Pulp database connection issue.")
> > +      end
> > +
> > +      if json['messaging_connection'] && json['messaging_connection']['connected'] != true
> > +        json['errors']['messaging_connection'] = _("Pulp message bus connection issue.")
> > +      end
> > +
> > +      json
> > +    end
>
> I was actually thinking of doing the other way around (e.g., to create a
> separate PR to refactor
> https://github.com/Katello/katello/blob/master/app/models/katello/ping.rb#L105
> to use this service class) [image: :smile:]
> Let me know which way you'd prefer.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Katello/katello/pull/5716/files#r50104629>.
>



-- 
Eric D. Helms
Red Hat Engineering
Ph.D. Student - North Carolina State University